### ISSUE TITLE
Routing.alert

### ISSUE THREAD

**Issue by s-cork** [anvilistas/anvil-extras#131]
See forum post. 

https://anvil.works/forum/t/how-to-close-alerts-when-navigating-away-from-current-page-with-hashrouting/9711/

Correct behaviour should probably be to close alert if the alert is dismissible and to prevent unloading if it's not dismissible.

**Comment by stenci** [comment]
I like Jay's decorator.
But perhaps, if implemented from inside the routing module, this shouldn't require a decorator.

Is it possible to replace the definitions of alert and confirm on all the forms, including the ones that don't import routing?
I don't think it is possible.

Then is it possible to do what the decorator does (modify the behavior of alert and confirm) just by importing routing?
It should be possible, why not?

If it is possible, then no decorators would be needed:
- the forms that are routed already have a decorator, and that one could do the job of managing open alert/confirm dialogs
- the forms that are not routed will just need to import routing

**Comment by s-cork** [comment]
@stenci - see #132 

which has the api
```python
from anvil_extras.routing import alert, confirm
alert('hello')
# or equivalently
from anvil_extras import routing
routing.alert('hello')
```

I don't think it's right for routing to implicitly override alerts everywhere.
But easily achieved in one's own app by doing something like this in a startup form/module

```python
import anvil
from anvil_extras import routing
anvil.alert = routing.alert
anvil.confirm = routing.confirm
```

**Comment by stenci** [comment]
I replaced all the calls to my `alert2` on the app I'm working on, copied the routing module with its new alert management and it seems to be working fine.

I have 2 comments:

1. (We should address this if possible) The tooltip with the function signature and the code completion don't work. 

   A workaround is to comment the `anvil.alert = routing.alert` and reload the IDE page. It's an ugly workaround, because you would need to comment this out on the startup form when you are editing another module.
This is a problem because you always need to refer to the documentation to know what are the available arguments.

2. (Not a big deal) Doing the import and redefining `alert` and `confirm` in the startup module works, but it would need to be before the `from anvil import *`, which by default appears on top.

   Basically we need to do this in the startup form:
```python
import anvil
from anvil_extras import routing
anvil.alert = routing.alert
anvil.confirm = routing.confirm

from ._anvil_designer import MainTemplate
from anvil import *
import anvil.server
...
```

Pros: It works. You do it once in the startup form and it affects all the forms.
Cons: Ain't pretty!

**Comment by s-cork** [comment]
Do you not get any code completions? I get this:
<img width="545" alt="Screenshot 2021-09-11 at 01 26 49" src="https://user-images.githubusercontent.com/36813890/132893592-93124045-6113-44fe-9f51-9f49526d0bfa.png">

and can adjust the docstring to be better.
Unfortunately anvil doesn't take into account the types of user defined parameters yet.

I'm happy to be out voted
what's the prettier option here?

**Comment by s-cork** [comment]
actually i might have a new plan to get around this - will keep you posted

**Comment by s-cork** [comment]
ok @stenci do you wanna take a look at that branch now in #132 
Just use anvil alerts as normal. 
don't worry about anything.
